feat: System Health Report

ankush opened the pull request:
Screenshot speaks louder than words.

![Screen Shot 2024-04-19 at 17 13 16-fullpage](https://github.com/frappe/frappe/assets/9079960/3e8c1fa9-0337-472f-85ed-88dcbd0c6f9d)


TODO:
- [x] perf / caching ( removed most slow funcs, should load fine in <30 seconds in most cases, can make separate small requests in future if required)
- [x] error handling
- [x] tabs -> section (less clicking around)
- [x] highlight "bad" indicators
- [x] constantly failing scheduled jobs (can reveal things like broken backup, repost etc)
- [x] highlight error with >10 occurrences


Closes https://github.com/frappe/frappe/issues/25486

ankush committed:
feat: System Health Report

ankush committed:
feat: background worker monitoring

ankush committed:
feat: better bench doctor in UI

ankush committed:
feat: socketio health check

ankush committed:
feat: email health checks

ankush committed:
feat: Errors in System Health

ankush committed:
feat: database health stats

ankush committed:
feat: cache health

ankush committed:
feat: backup health

ankush committed:
feat: system health - users

ankush commented:
I don't like tabs for this TBH, too much clicking around :moyai:

ankush committed:
refactor: Single page instead of tabs

ankush committed:
feat: background jobs test

ankush committed:
fix: exception handling for health report

ankush committed:
feat: highlight bad indicators

ankush committed:
chore: rename child doctypes

ankush committed:
fix(UX): help links and relative URLs

also closes https://github.com/frappe/frappe/issues/23020

ankush committed:
feat: extend highlight to child tables

ankush committed:
refactor: use table for errors

rmehta commented:
Nice, make add some emphasis, colour for measures that seem out of some range?

ankush commented:
@rmehta red borders are for that 🙈

ankush commented:
Idea for more metrics:

- [ ] constantly failing scheduled jobs (can reveal things like broken backup, repost etc)

rmehta commented:
@ankush ah - that is mixed up with mandatory (value missing). Maybe try something else, like text in  bold and ping / yellow background?

ankush committed:
feat: failng scheduled jobs

ankush committed:
refactor: misc

- fix styles
- hardcode perm check
- few more indicators
- cache directory size for 5 min (rapid refreshes should be fast enough)

git-avc commented:
Maybe something like this?
![image](https://github.com/frappe/frappe/assets/94137451/133d56c9-6fe8-41a3-999f-ff41e3d41f10)

ankush commented:
I changed it to light red background (almost pink) + bold fonts. 

Not updating screenshot, that takes some time to produce fake data :woozy_face:
